src/controllers/expenses.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const {
+  getAllExpenses,
+  addExpense,
+  getExpense,
+  deleteExpense,
+  updateExpense,
+} = require('../services/exprensesService');
+const { getAllUsers } = require('../services/usersServer.js');
+
+async function getAllExpensesAction(req, res) {
+  const expenses = await getAllExpenses();
+
+  res.json(expenses);
+};
+
+async function addExpenseAction(req, res) {
+  const expense = req.body;
+  const allowedKeys = [
+    'title',
+    'amount',
+    'category',
+    'note',
+    'userId',
+  ];
+  const keysExpense = Object.keys(expense);
+
+  const missingFields = keysExpense.every(key => allowedKeys.includes(key));
+  const users = await getAllUsers();
+
+  if (
+    !users.length
+    || !users.filter((el) => el.id === expense.userId).length
+    || !missingFields) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = await addExpense(expense);
+
+  res.json(newExpense);
+  res.status(201);
+};
+
+async function getExpenseAction(req, res) {
+  const { id } = req.params;
+  const expenses = await getAllExpenses();
```

Comment:
same question. why don't use  `getExpense`?

Reply:
fix it for all cases

Reply:
done